core/modules/file/views/views_handler_field_file_icon.inc

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @file
+ * Definition of views_handler_field_file_icon.
+ */
+
+/**
+ * @ingroup views_field_handlers
+ */
+class views_handler_field_file_icon extends views_handler_field {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function option_definition() {
+    $options = parent::option_definition();
+    $options['icon_size'] = array('default' => 32);
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function options_form(&$form, &$form_state) {
+    parent::options_form($form, $form_state);
+    $form['icon_size'] = array(
+      '#type' => 'number',
+      '#title' => t('Icon size'),
+      '#min' => 8,
+      '#max' => 300,
+      '#default_value' => $this->options['icon_size'],
+      '#field_suffix' => 'px',
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render($values) {
+    $value = $this->get_value($values);
+
+    // If there is an extension.
+    if ($extension = pathinfo($value, PATHINFO_EXTENSION)) {
+      $options = array(
+        'attributes' => array(
+          'width' => $this->options['icon_size'],
+          'height' => $this->options['icon_size'],
+        ),
+      );
+      $icon_name = 'file-' . $this->mapExtension($extension);
+      $icon = icon($icon_name, $options);
+      // Fall back to generic file icon.
+      if (strpos($icon, '<!-- Icon') === 0) {
```

Comment:
The word `Icon` is passed through `t()` in the `icon()` function. Wouldn't that cause this check to fail in languages other than English then? 🤔

Reply:
...it would have been easier if `icon()` was returning `FALSE` if the icon is not found.

Reply:
...perhaps try using `icon_get_path()` instead for this check, since it returns `NULL`.

Reply:
This is only a *fallback* that hardly ever strikes.

Many thanks for the `t()` hint - I'll take a look.

(It makes sense, that the icon() function  *always* returns a string. Really. :wink:)